apis/basic/composition.yaml

```
@@ -6,528 +6,272 @@ metadata:
+            {{- end }}
+            {{- define "region" }}
+              region: {{ .observed.composite.resource.spec.parameters.region }}
+            {{- end }}
+            {{- define "deletionPolicy" }}
+              deletionPolicy: {{ .observed.composite.resource.spec.parameters.deletionPolicy | default "Delete" }}
+            {{- end }}
 
-          - name: subnetPublicA
-            base:
-              apiVersion: ec2.aws.upbound.io/v1beta1
-              kind: Subnet
-              metadata:
-                labels:
-                  access: public
-              spec:
-                forProvider:
-                  cidrBlock: 192.168.0.0/18
-                  mapPublicIpOnLaunch: true
-                  tags:
-                    kubernetes.io/role/elb: "1"
-                  vpcIdSelector:
-                    matchControllerRef: true
-            patches:
-              - type: PatchSet
-                patchSetName: providerConfigRef
-              - type: PatchSet
-                patchSetName: deletionPolicy
-              - type: PatchSet
-                patchSetName: network-id
-              - type: PatchSet
-                patchSetName: network-id-tags
-              - type: PatchSet
-                patchSetName: region
-              - type: FromCompositeFieldPath
-                fromFieldPath: spec.parameters.region
-                toFieldPath: spec.forProvider.availabilityZone
-                transforms:
-                  - type: string
-                    string:
-                      fmt: '%sa'
-                      type: Format
-              - type: FromCompositeFieldPath
-                fromFieldPath: spec.parameters.region
-                toFieldPath: metadata.labels[zone]
-                transforms:
-                  - type: string
-                    string:
-                      fmt: '%sa'
-                      type: Format
-              - type: ToCompositeFieldPath
-                fromFieldPath: metadata.annotations[crossplane.io/external-name]
-                toFieldPath: status.subnetIds[0]
-              - type: ToCompositeFieldPath
-                fromFieldPath: metadata.annotations[crossplane.io/external-name]
-                toFieldPath: status.publicSubnetIds[0]
+            apiVersion: ec2.aws.upbound.io/v1beta1
+            kind: VPC
+            metadata:
+              labels:
+                {{- include "networkId" . | nindent 4 }}
+              annotations:
+                {{ setResourceNameAnnotation ("vpc") }}
+            spec:
+              forProvider:
+                cidrBlock: {{ $params.vpcCidrBlock }}
+                enableDnsHostnames: true
+                enableDnsSupport: true
+                tags:
+                  Name: {{ .observed.composite.resource.metadata.name }}
+                {{- include "region" . | nindent 2 }}
+              {{- include "providerConfigRef" . }}
```

```suggestion
              {{- include "providerConfigRef" $params }}
```
Let's use $params consistently across all includes in a similar way to this suggestion